drivers/overlay/encryption.go

```diff
@@ -224,7 +225,9 @@ func programSA(localIP, remoteIP net.IP, spi *spi, k *key, dir int, add bool) (f
 	if add {
 		action = "Adding"
 		xfrmProgram = ns.NlHandle().XfrmStateAdd
-		crypt = &netlink.XfrmStateAlgo{Name: "cbc(aes)", Key: k.value}
+		salt := make([]byte, 4)
+		binary.LittleEndian.PutUint32(salt, k.tag)
```

Comment:
I don't see where `k.tag` is initialized. This should be a random value, per the above.